plugins/idea/src/main/java/org/robovm/idea/running/RoboVmIOSRunConfigurationSettingsEditor.java

```diff
@@ -382,6 +459,270 @@ private void updatePairedWatch(boolean moduleChanged, Boolean valueToSet) {
+    }
+
+    private boolean isSigningIdentityValid(SigningIdentity signingIdentity, Config config, boolean createErrorStrings) {
+        if (createErrorStrings) {
+            signingIdentityErrors.clear();
+        }
+
+        boolean currentlyValid = false;
+
+
+        Decorator<ProvisioningProfile> currentSelectedProvisioningProfile = Decorator.from(provisioningProfile);
+
+        boolean isProfileSelectedAndValid = currentSelectedProvisioningProfile != null && currentSelectedProvisioningProfile.data != null;
+        if (isProfileSelectedAndValid) {
+            isProfileSelectedAndValid = isProvisioningProfileValid(currentSelectedProvisioningProfile.data, config, false);
+        }
+
+        if (isProfileSelectedAndValid) {
+            Set<String> certFingerprints = currentSelectedProvisioningProfile.data.getCertFingerprints();
+            if (certFingerprints.contains(signingIdentity.getFingerprint())) {
+                currentlyValid = true;
+            } else {
+                if (createErrorStrings) {
+                    signingIdentityErrors.add("Signing Certificate is not in the selected Provisioning Profile");
+                }
+            }
+        } else { //Check all of them
+            for (ProvisioningProfileDecorator provisioningProfile : provisioningProfiles) {
+                ProvisioningProfile profile = provisioningProfile.data;
+                if (profile != null) {
+                    if (isProvisioningProfileValid(profile, config, false)) {
+                        Set<String> certFingerprints = profile.getCertFingerprints();
+                        if (certFingerprints.contains(signingIdentity.getFingerprint())) {
+                            currentlyValid = true;
+                        }
+                    }
+                }
+            }
+
+            if (!currentlyValid) {
+                if (createErrorStrings) {
+                    signingIdentityErrors.add("Signing Certificate is not in any of the Provisioning profiles valid for this app");
+                }
+            }
+        }
+
+
+
+        return currentlyValid;
+    }
+
+    private boolean isProvisioningProfileValid (ProvisioningProfile profile, Config config, boolean createErrorStrings) {
+        if (createErrorStrings) {
+            provisioningProfileErrors.clear();
+        }
+
+        String appIdPrefix = profile.getAppIdPrefix();
+        String appId = profile.getAppId();
+        String appIdNoPrefix = appId.split(appIdPrefix + ".")[1];
+
+        boolean currentlyCompatible = true;
+
+        //Check the id
+        if (!bundleLabel.getText().equalsIgnoreCase(appIdNoPrefix)) {
+            currentlyCompatible = false;
+            if (createErrorStrings) {
+                provisioningProfileErrors.add("Bundle ID from profile=[" + appIdNoPrefix + "] does not match module Bundle ID [" + bundleLabel.getText() + "]");
+            }
+        }
+
+        //Check if required entitlements are in the profile
```

Comment:
I'm not big entitlement expert. but I'm not sure if all items from entitlement.plist expected to be present in plist.